sfge/src/main/java/com/salesforce/graph/ops/MethodUtil.java

```diff
@@ -645,38 +645,66 @@ public static List<ApexPath> getPaths(
                 // current class or a static
                 // method on a another class
                 String definingType;
-                final String methodName = methodCallExpression.getMethodName();
-                String fullMethodName = methodCallExpression.getFullMethodName();
-                if (methodName.equals(fullMethodName)) {
-                    // The method is being called on a class onto itself
-                    definingType = vertex.getDefiningType();
+                List<String> potentialDefiningTypes = new ArrayList<>();
+                if (methodCallExpression.isThisReference()) {
+                    // A call to `this.someMethod()` means the method must be on the defining type.
+                    definingType = methodCallExpression.getDefiningType();
+                    potentialDefiningTypes.add(
+                            ApexStandardLibraryUtil.getCanonicalName(definingType));
+                } else if (methodCallExpression.isEmptyReference()) {
+                    // For an empty reference (e.g., `someMethod()` as opposed to
+                    // `whatever.someMethod()`), the first place we should look is
+                    // the call's own defining type.
+                    definingType = methodCallExpression.getDefiningType();
+                    potentialDefiningTypes.add(
+                            ApexStandardLibraryUtil.getCanonicalName(definingType));
+                    // Inner classes can also invoke their outer class's static methods with an
+                    // empty reference, so if this is an inner class, we should check the outer
+                    // class too.
+                    if (definingType.contains(".")) {
+                        potentialDefiningTypes.add(definingType.split("\\.")[0]);
+                    }
+                } else if (methodCallExpression.isSuperVariableReference()) {
```

Comment:
This part isn't directly related to the bug in question, but prevents it from breaking any of the existing tests. (It also technically fixes a few that were already broken and disabled, which is nice. Might let us DNR another bug.)